File: solutions/Leetcode_102/leetcode_102.py
class Solution1:
    def level_order(self, root):
        if not root:
            return []
        queue = [root]
        traversal = []
        while queue:
            level_values = []
            for i in range(len(queue)):
                node = queue.pop()
                level_values.append(node.val)
                if node.left:
                    queue.insert(0, node.left)
                if node.right:
                    queue.insert(0, node.right)
            traversal.append(level_values)
        return traversal

File: solutions/Leetcode_102/leetcode_102.cc
#include <queue>
#include <vector>

using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution1
{
public:
    vector<vector<int>> levelOrder(TreeNode *root)
    {
        if (root == nullptr)
            return {};
        queue<TreeNode *> q;
        q.push(root);
        vector<vector<int>> traversal;
        int num_in_level = 0;
        TreeNode *node;
        while (!q.empty())
        {
            num_in_level = q.size();
            vector<int> level;
            for (int i = 0; i < num_in_level; i++)
            {
                node = q.front();
                q.pop();
                level.push_back(node->val);
                if (node->left != nullptr)
                    q.push(node->left);
                if (node->right != nullptr)
                    q.push(node->right);

            }
            traversal.push_back(level);
        }
        return traversal;
    }
};
